src/leaflet-search.js

```diff
@@ -177,30 +176,37 @@ L.Control.Search = L.Control.extend({
 	// 			this.setLayer(e.layer);
 	// },
 
-	setLayer: function(layer) {	//set search layer at runtime
+  setLayer: function(layers) { //set search layer at runtime
 		//this.options.layer = layer; //setting this, run only this._recordsFromLayer()
-		this._layer = layer;
-		this._layer.addTo(this._map);
+    var map = this._map;
+    if (layers.searchGroup) {
```

Comment:
@GIGI-Cp  the best way would be check is layers is an array... if array then use loop